libqtile/bar.py

```diff
@@ -275,6 +278,12 @@ def _remove_crashed_widgets(self):
     def finalize(self):
         self.drawer.finalize()
 
+    def kill_window(self):
+        """Kill the window when the bar's screen is no longer being used."""
+        self.drawer.finalize()
+        self.window.kill()
+        self.window = None
```

Comment:
Might be a dumb question, but why not put these in `finalize` and call that instead?

Reply:
Finalize is currently only for very final finalising, rather than shutting things down. Although it'd be easy to argue "well we don't need both of those functionalites here, so why not just use finalize?" but this functionality is currently needed in another PR https://github.com/qtile/qtile/pull/2360/commits/8c231cbd44d8d527623415718ccb1531fbd83384#diff-effdc1f9379ed88ce401e42f5852cc8db7444a9b885995477d68709069ca4977R276-R280